### PR TITLE
Fixed failed NetCore conversions

### PR DESCRIPTION
**Pull request by MiYanni**
## Description

This PR does these things:
https://github.com/Azure/azure-powershell/issues/5719
https://github.com/Azure/azure-powershell/issues/5753
https://github.com/Azure/azure-powershell/issues/5623
https://github.com/Azure/azure-powershell/issues/5620

Basically, this allows the rest of the Commands projects to build in .NET Core. No test projects are involved in this. No guarantee that the cmdlets actually work... yet.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by MiYanni** [comment]
Core sign: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign/19/
Framework sign: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/288/

File: src/ResourceManager/AzureBackup/Commands.AzureBackup/Cmdlets/Restore/RestoreAzureRMBackupItem.cs [hunk 1]
```diff
@@ -50,8 +50,7 @@ public override void ExecuteCmdlet()
                     Region = RecoveryPoint.Location,
                 };
 
-                JavaScriptSerializer serializer = new JavaScriptSerializer();
-                string azureIaaSVMRecoveryInputsCSMObjectString = serializer.Serialize(azureIaaSVMRecoveryInputsCSMObject);
+                string azureIaaSVMRecoveryInputsCSMObjectString = JsonConvert.SerializeObject(azureIaaSVMRecoveryInputsCSMObject);
```
Comment (markcowl) [inline src/ResourceManager/AzureBackup/Commands.AzureBackup/Cmdlets/Restore/RestoreAzureRMBackupItem.cs]: Check this (and similar instances with the RecoveryServices team)

File: src/ResourceManager/AzureBackup/Commands.AzureBackup/packages.config [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="Microsoft.Azure.Management.BackupServices" version="1.0.5-preview" targetFramework="net45" />
-</packages>
+  <package id="Microsoft.WindowsAzure.Management.Scheduler" version="6.0.0" targetFramework="net452" />
```
Comment (markcowl) [inline src/ResourceManager/AzureBackup/Commands.AzureBackup/packages.config]: Check this dependency

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Models/FileSystemCmdletProviderEncoding.cs [hunk 3]
```diff
@@ -12,24 +12,23 @@
 // limitations under the License.
 // ----------------------------------------------------------------------------------
 
-using Microsoft.Azure.Commands.Dns.Services;
```
Comment (markcowl) [inline src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Models/FileSystemCmdletProviderEncoding.cs]: string diff

File: src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/Common/PSAsrClient.cs [hunk 4]
```diff
@@ -147,7 +148,7 @@ public partial class PSRecoveryServicesClient
                 2);
             cikTokenDetails.PropertyBag = new Dictionary<string, object>();
 
-            var shaInput = new JavaScriptSerializer().Serialize(cikTokenDetails);
+            var shaInput = JsonConvert.SerializeObject(cikTokenDetails);
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/Common/PSAsrClient.cs]: RecoveryServices team should look at this

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Common/PSSiteRecoveryClient.cs [hunk 5]
```diff
@@ -275,7 +276,7 @@ public string GenerateAgentAuthenticationHeader(string clientRequestId)
             cikTokenDetails.Version = new Version(1, 2);
             cikTokenDetails.PropertyBag = new Dictionary<string, object>();
 
-            string shaInput = new JavaScriptSerializer().Serialize(cikTokenDetails);
+            string shaInput = JsonConvert.SerializeObject(cikTokenDetails);
```
Comment (markcowl) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Common/PSSiteRecoveryClient.cs]: Have the team take a look at this serialization change

File: src/ResourceManager/AzureBackup/Commands.AzureBackup/packages.config (hunk 2 above)
Comment (MiYanni) [reply src/ResourceManager/AzureBackup/Commands.AzureBackup/packages.config]: This dependency is required. Going to create an issue for removing it.

File: src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Models/FileSystemCmdletProviderEncoding.cs (hunk 3 above)
Comment (MiYanni) [reply src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Models/FileSystemCmdletProviderEncoding.cs]: This seems to be some kind of GitHub diff issue. Looking at the Git log for the file, there are only 2 commits in the history. 1 for creating it and another when I added the comment linking to the GitHub issue I created for it.

File: src/ResourceManager/AzureBatch/Commands.Batch/Models/BatchClient.ApplicationPackages.cs [hunk 6]
```diff
@@ -117,7 +117,11 @@ public virtual PSApplicationPackage GetApplicationPackage(string resourceGroupNa
             try
             {
                 CloudBlockBlob blob = new CloudBlockBlob(new Uri(storageUrl));
+#if NETSTANDARD 
+                blob.UploadFromFileAsync(filePath).RunSynchronously();
```
Comment (markcowl) [inline src/ResourceManager/AzureBatch/Commands.Batch/Models/BatchClient.ApplicationPackages.cs]: need to fix

File: src/ResourceManager/DataFactories/Commands.DataFactories/Models/DataFactoryClient.DataSlices.cs [hunk 7]
```diff
@@ -151,7 +151,11 @@ public virtual void DownloadFileToBlob(BlobDownloadParameters parameters)
             CloudBlobContainer sascontainer = sourceClient.GetContainerReference(rli.Container);
             CloudBlobDirectory sourceDirectory = sascontainer.GetDirectoryReference(rli.Directory);
 
+#if NETSTANDARD
+            var bloblist = sourceDirectory.ListBlobsSegmentedAsync(true, BlobListingDetails.None, null, null, null, null).Result.Results;
```
Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactories/Models/DataFactoryClient.DataSlices.cs]: check

File: src/ResourceManager/DataFactories/Commands.DataFactories/Models/DataFactoryClient.DataSlices.cs [hunk 8]
```diff
@@ -177,8 +181,12 @@ public virtual void DownloadFileToBlob(BlobDownloadParameters parameters)
                 }
                 // adding _log suffix to differentiate between files and folders of the same name. Azure blob storage only knows about blob files. We could use nested folder structure
                 // as part of the blob file name and thus it is possible to have a file and folder of the same name in the same location which is not acceptable for Windows file system
-                string fileToDonwload = destBlob.Name.Remove(0, rli.Directory.Length);
-                destBlob.DownloadToFile(downloadFolderPath + fileToDonwload + "_log", FileMode.Create);
+                string fileToDownload = destBlob.Name.Remove(0, rli.Directory.Length);
+#if NETSTANDARD
+                destBlob.DownloadToFileAsync(downloadFolderPath + fileToDownload + "_log", FileMode.Create).RunSynchronously();
```
Comment (markcowl) [inline src/ResourceManager/DataFactories/Commands.DataFactories/Models/DataFactoryClient.DataSlices.cs]: check

**Pull request merged**